### ISSUE TITLE
[css-sizing-4] How does a max-height affect the inline size computed through an aspect ratio

### ISSUE THREAD

**Issue by cbiesinger** [w3c/csswg-drafts#5257]
https://drafts.csswg.org/css-sizing-4/#aspect-ratio

The title is not very descriptive, sorry -- I couldn't think of a succinct way to describe the issue. Basically:

```html
  <div style="aspect-ratio: 1/1; max-height: 50px; font: 10px/1 Ahem; float: left;">
    XX XXX <!-- max-size 60px, min-size: 30px -->
  </div>
```
So we have shrink-to-fit sizing in the inline axis and an aspect ratio size in the block axis. Since we have a lot of space, inline size is 60px (max intrinsic size), and height is 60px.

BUT, max-height is 50px. Should we:
- Clamp max-height to 50px and recompute the inline size based on that (here, 50px), like we do for <img>?
http://plexode.com/eval3/#s=aekVQXANJVQMbAx1yAXgePQOCGwESERFRWRwBg52foVmaHz1PAQEdSk5IAVRTRJmPVVVRVBsQEFdKQg9RTUJERklQTUVGUw9EUE4QFKCal7lOQlkOpp6gBhMWHKutHRByHwNeAA==

- Not do this complex thing and just let this be sized to 60x50?

Additional question if we do option one: Should min-block-size be taken into account too? If yes, auto should probably resolve to 0 for the purpose of inline sizing.

cc @bfgeek @fantasai

**Comment by tabatkins** [comment]
Yes, the max-height is definite, so it should get transferred across the ratio, and give a max-width of 50px, exactly like the image example. Unfortunately the current [aspect ratio constraint rules](https://www.w3.org/TR/CSS2/visudet.html#min-max-widths) are written with the assumption that the aspect ratio of the actual content and the used aspect ratio are identical... so we need to write up some explicit rules about this. :/ (Whereas in this example, the AR of the actual content is `6 / 1`, different from the desired ratio of `1 / 1`.)

If we *don't* transfer this, then max-size properties basically don't interact with aspect-ratio at all, which sucks.

**Comment by fantasai** [comment]
OK, added some prose to the spec to describe the transfer of sizing constraints. Let us know if it looks correct? :)

**Comment by cbiesinger** [comment]
@fantasai I added some comments on https://github.com/w3c/csswg-drafts/commit/ede56bdbd7c3d365b0086592f35c13a93072b2ce

**Comment by fantasai** [comment]
@cbiesinger Replied in https://github.com/w3c/csswg-drafts/commit/ede56bdbd7c3d365b0086592f35c13a93072b2ce#commitcomment-40570230 and added https://github.com/w3c/csswg-drafts/commit/a6fc47e51e9806b730c258c6775ab205bda8cd48

**Comment by cbiesinger** [comment]
fwiw, I did add a test for this in https://github.com/web-platform-tests/wpt/commit/5938e6771f16b91f2da43291e8e2c424cbb14432; I would appreciate a spec editor reviewing it but I'm not sure this needs the WPT label.

I also added another comment on the commit.

**Comment by fantasai** [comment]
@cbiesinger Commented on the changeset... the test seems really off?
